### ISSUE TITLE
Respect the XDG base directory specification

### ISSUE THREAD

**Issue by dsifford** [prettier/prettier#4810]
Referred here from an [issue that I opened in cosmiconfig](https://github.com/davidtheclark/cosmiconfig/issues/152#issuecomment-402547011).

That issue was pretty detailed so, to save time, below I'll paste it again in full:

---

Hello,

I searched in the issues here and shockingly I haven't found this mentioned.

Currently, cosmiconfig searches up the parent tree from the working directory, stopping at `$HOME`, which is fine and most, I would assume, are okay with this.

However, for filesystem psychopaths like myself, it pains me to have to move any sort of user configuration file out `$XDG_CONFIG_HOME/project-name/` directory (usually, `~/.config/project-name/`)  and pollute my home directory.

Any thoughts on adding this as a final search path?

You can read up on the specification [here](https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html) and [here](https://wiki.archlinux.org/index.php/XDG_Base_Directory_support#XDG_Base_Directory_specification).

In short, using prettier as an example, I'd imagine the workflow to go like this...

1. Perform the current lookup traversal as it is done right now.
2. If nothing found, check the following:
  - `$XDG_CONFIG_HOME/prettier/config`
  - `$XDG_CONFIG_HOME/prettier/[...current standard file name checks]`
 
---

The cosmiconfig maintainers stated that this would be easily possible using the [searchplaces](https://github.com/davidtheclark/cosmiconfig/blob/master/README.md#searchplaces) API.

Thanks for considering.

**Comment by ELLIOTTCABLE** [comment]
Just :+1: on this — although I *really* think this belongs upstream in cosmicconfig, so eh.

**Comment by masaeedu** [comment]
@ELLIOTTCABLE The cosmicconfig maintainers on the other hand seem to think it belongs here.

**Comment by thorn0** [comment]
Prettier isn't going to support any kind of global configuration. See https://github.com/prettier/prettier/issues/8094
